packages/transformers/babel/src/jsx.js

```diff
@@ -49,7 +52,14 @@ export default async function getJSXOptions(config: Config) {
 
   if (pragma || JSX_EXTENSIONS[path.extname(config.searchPath)]) {
     return {
-      plugins: [['@babel/plugin-transform-react-jsx', {pragma, pragmaFrag}]]
+      plugins: [
+        ['@babel/plugin-transform-react-jsx', {pragma, pragmaFrag}],
+        reactLib === 'react' &&
+          (await options.packageManager.resolve(
```

Comment:
There need to be check here if the `react-refresh-transformer` is running or this needs to be executed separately in that transformer.